syntax = "proto3";

import "Task.proto";

message AddTaskRequest {
  model.Task task = 1;
}

message AddSubTaskRequest {
  model.Task task = 1;
  model.TaskId parent_id = 2;
}

message EditTaskRequest {
  model.TaskId id = 1;
  model.Task task = 2;
}

message DeleteTaskRequest {
  model.TaskId id = 1;
}

message CompleteTaskRequest {
  model.TaskId id = 1;
}

message ShowByLabelRequest {
  string label = 1;
  model.TasksSortBy sort_by = 2;
}

message ShowParentsRequest {
  model.TasksSortBy sort_by = 1;
}

message ShowTaskRequest {
  model.TaskId id = 1;
  model.TasksSortBy sort_by = 2;
}

message ShowAllRequest {
  model.TasksSortBy sort_by = 1;
}

message SaveRequest {
  string filename = 1;
}

message LoadRequest {
  string filename = 1;
}
